tests/integrations/base_tester.py

```diff
@@ -0,0 +1,245 @@
+        # The raw config can contain multiple command stages (train, export, deploy)
+        raw_config = yaml.safe_load(config_path)
+        # Remove cadence for easier processing. It's saved, but not utilized later
+        self.cadence = raw_config.pop("cadence")
+
+        # Will be used in tests to skip test if type doesn't apply
+        self.command_types = [_type for _type in raw_config]
+
+        # 3 The command stub is fetched. See method description
+        self.command_stubs_final = self.get_base_commands(raw_config)
+
+        # 3 extract test args.
+        self.test_args = self.get_test_args(raw_config)
+
+        # Command-specific args are loaded into their respective pydantic classes.
+        # pre_args are CLI commands that can precede the python call
+        # e.g. "torch distributed launch" or env variable setting. If any are present,
+        # they are read in as well.
+        self.configs = {
+            _type: {
+                "pre_args": config.pop("pre_args", None),
+                "args": self.command_args_classes[_type](config.get("command_args")),
+            }
+            for _type, config in raw_config
+        }
+        # Generate full CLI commands for each stage of run
+        self.commands = self.compose_command_scripts(self.configs)
+        # 3 Override to save any information which may be needed for testing post-run
+        self.capture_pre_run_state()
+        # All commands are run in order
+        self.run_commands(self.commands)
+        yield  # all tests are run here
+        # 3 WIP post-test teardown
+        self.cleanup_files()
+        self.check_file_creation()
+        # pytest.skip(f"Not a {class_command_type} command")
+
+    @classmethod
+    def get_base_commands(cls, configs: Dict[str, Union[str, BaseModel]]):
+        """
+        Returns the command stubs to use. e.g. "sparseml.yolov5.train".
+        If the stub for the integration needs to be determined based on the configs
+        (e.g. sparseml.transformers.{task}.train), ovveride this function
+
+        :param configs: unprocessed configs dict
+        :return: dict of type {"command_type": "command_stub"}
+        """
+        return cls.command_stubs
+
+    @classmethod
+    def get_test_args(cls, configs):
+        """
+        Return test args. Test args are open ended in function and are meant to
+        interact with the tests.Example target args are "target F1 score" and
+        "flag to not test onnx model structure"
+
+        WIP, but there may not be much here than can be generalizable. i.e. this
+        function my have to be implemented on the integration level
+        """
+        return None
+
+    @classmethod
+    def compose_command_scripts(cls, configs):
+        """
+        For each command, create the full CLI command by combining the pre args,
+        command stub, and run args.
+        """
+        commands = []
+        for _type, config in configs:
```

Comment:
```suggestion
        for _type, config in configs.items():
```
`configs` is a dictionary right?